src/JMS/Serializer/Handler/DateHandler.php

```diff
@@ -173,6 +191,18 @@ private function parseDateTime($data, array $type, $immutable = false)
         return $datetime;
     }
 
+    private function parseDateInterval($data)
+    {
+        $dateInterval = null;
+        try {
+            $dateInterval = new \DateInterval($data);
+        } catch (\Exception $e) {
+            throw new RuntimeException(sprintf('Invalid dateinterval "%s", expected ISO 8601 format', $data));
```

Comment:
I was just copying the one example of catching and rethrowing an exception within the project to keep things consistent, but I'm happy to add that as a 3rd param. It makes sense.